packages/starlight/utils/navigation.ts

```diff
@@ -134,8 +135,19 @@ function makeLink(
 	badge?: Badge,
 	attrs?: LinkHTMLAttributes
 ): Link {
-	if (!isAbsolute(href)) href = pathWithBase(href);
-	const isCurrent = href === ensureTrailingSlash(currentPathname);
+
+	if (isAbsolute(href))
+		return { type: 'link', label, href, isCurrent: false, badge, attrs: attrs ?? {} };
+
+	href = pathWithBase(href);
+	const isCurrent = href === ensureTrailingSlash(stripExtension(currentPathname));
+
+	if (project.build.format === 'file') {
+		// edge case for index.html
+		if (href === '/') href = '/index/';
+		href = stripTrailingSlash(href) + '.html';
+	}
```

Comment:
Is there a way to extract some of this into a reusable function that formats any relative path according to Astro’s config?

We’ll need to do the same for the `<SiteTitle />` component which currently does this:

https://github.com/withastro/starlight/blob/5499c2524ca958c6a59581eaf6184e14c4fccecf/packages/starlight/components/SiteTitle.astro#L7

Would also make it easier to test because then we can just have a bunch of tests for that function passing different arguments and checking it returns what we expect.

Reply:
I didn't know it was used somewhere else. Yes, I can give it a shot.